test/policies/proxy/proxy.test.js

```diff
@@ -0,0 +1,173 @@
+const path = require('path');
+const fs = require('fs');
+const request = require('supertest');
+const assert = require('chai').assert;
+
+const config = require('../../../lib/config');
+const gateway = require('../../../lib/gateway');
+const { findOpenPortNumbers } = require('../../common/server-helper');
+
+const originalGatewayConfig = config.gatewayConfig;
+
+const serverKeyFile = path.join(__dirname, '../../fixtures/certs/server', 'server.key');
+const serverCertFile = path.join(__dirname, '../../fixtures/certs/server', 'server.crt');
+const invalidClientCertFile = path.join(__dirname, '../../fixtures', 'agent1-cert.pem');
+const clientKeyFile = path.join(__dirname, '../../fixtures/certs/client', 'client.key');
+const clientCertFile = path.join(__dirname, '../../fixtures/certs/client', 'client.crt');
+const chainFile = path.join(__dirname, '../../fixtures/certs/chain', 'chain.pem');
+
+let backendServerPort;
+
+function expectedResponse (app, status, contentType) {
+  return request(app).get('/endpoint').expect(status).expect('Content-Type', contentType);
+}
+
+describe('proxy policy', () => {
+  const defaultProxyOptions = {
+    target: {
+      keyFile: clientKeyFile,
+      certFile: clientCertFile,
+      caFile: chainFile
+    }
+  };
+  let app, backendServer;
+
+  before('start HTTP server', (done) => {
+    findOpenPortNumbers(1).then((ports) => {
+      const https = require('https');
+      const express = require('express');
+      const expressApp = express();
+
+      backendServerPort = ports[0];
+
+      expressApp.all('*', function (req, res) {
+        res.status(200).json();
+      });
+
+      backendServer = https.createServer({
+        key: fs.readFileSync(serverKeyFile),
+        cert: fs.readFileSync(serverCertFile),
+        ca: fs.readFileSync(chainFile),
+        requestCert: true,
+        rejectUnauthorized: true
+      }, expressApp);
+
+      backendServer.listen(backendServerPort, done);
+    });
+  });
+
+  after('clean up', (done) => {
+    config.gatewayConfig = originalGatewayConfig;
+    backendServer.close(done);
+  });
+
+  describe('proxyOptions', () => {
+    it('raises an error when incorrect TLS file paths are provided', (done) => {
+      const serviceOptions = { target: { keyFile: '/non/existent/file.key' } };
+
+      setupGateway(serviceOptions).catch(err => {
```

Comment:
Nip: this can be replaced with a single line: `should(setupGateway(serviceOptions)).be.rejectedWith('no such file or directory')`